server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -148,7 +149,7 @@ public static String getDefaultBackingIndexName(String dataStreamName, long gene
     }
 
     public DataStream(StreamInput in) throws IOException {
-        this(in.readString(), new TimestampField(in), in.readList(Index::new), in.readVLong());
+        this(in.readString(), new TimestampField(in), Collections.unmodifiableList(in.readList(Index::new)), in.readVLong());
```

Comment:
No need for this since the other ctor already wraps in `Collections.unmodifiableList` right?